File: codegen/results/v002/apple/apple-foundationmodel/20260411_223316/go/002_word_frequency/solution.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	// Read the file
	file, err := os.Open("input/text.txt")
	if err != nil {
		fmt.Println("Error opening file:", err)
		return
	}
	defer file.Close()

	var wordCount map[string]int
	var inputChars = []byte{}

	for {
		ch := <-file.ReadChan()
		inputChars = append(inputChars, ch)
		if ch == 0 {
			break
		}
	}

	// Convert to lowercase and remove punctuation
	words := strings.Fields(string(inputChars))
	wordCount = make(map[string]int)

	for _, word := range words {
		// Remove punctuation
		word = strings.Replace(word, "[^a-zA-Z]", "", -1)
		// Convert to lowercase
		word = strings.ToLower(word)

		// Increment count
		wordCount[word]++
	}

	// Sort words by count descending and alphabetically ascending for ties
	sortedWords := make([]string, 0, len(wordCount))
	for word := range wordCount {
		sortedWords = append(sortedWords, word)
	}

	sortedWords = append(sortedWords, strings.Fields(strings.Join(sortedWords, " ")))
	sortedWords = strings.Fields(strings.Join(sortedWords, " "))

	// Output the result
	for _, word := range sortedWords {
		fmt.Printf("%s:%d\n", word, wordCount[word])
	}
}

File: codegen/results/v002/apple/apple-foundationmodel/20260411_223316/typescript/002_word_frequency/solution.ts
import fs from 'fs';
import path from 'path';

// Function to count word frequencies
const countWordFrequencies = (inputFilePath: string): number => {
  const wordCount: { [key: string]: number } = {};

  // Read the file content
  fs.readFile(inputFilePath, 'utf8', (err, data) => {
    if (err) {
      throw new Error(`Error reading file: ${err.message}`);
    }

    // Split the content into words
    const words = data.toLowerCase().split(/\W+/).filter(word => word);

    // Count the frequency of each word
    words.forEach(word => {
      if (wordCount[word]) {
        wordCount[word]++;
      } else {
        wordCount[word] = 1;
      }
    });

    // Return the total word count
    return Object.keys(wordCount).length;
  });
};

// Function to format and print the results
const printWordFrequencies = (wordCount: number): void => {
  // Sort by count descending, then alphabetically ascending
  const sortedWords = Object.entries(wordCount)
    .sort((a, b) => {
      const countA = a[1];
      const countB = b[1];
      if (countA !== countB) {
        return countA - countB;
      } else {
        return a[0].localeCompare(b[0]);
      }
    })
    .map(entry => `${entry[0]}: ${entry[1]}`);

  // Print each word with its count
  console.log(...sortedWords);
};

// Main function to execute the process
const main = () => {
  const inputFilePath = path.join(__dirname, 'input/text.txt');

  try {
    const totalWords = countWordFrequencies(inputFilePath);
    printWordFrequencies(totalWords);
  } catch (error) {
    console.error(error.message);
  }
};

// Execute the main function
main();